tests/unit/query_plan_read_write_typecheck.cpp

```diff
@@ -253,3 +253,23 @@ TEST_F(ReadWriteTypeCheckTest, Foreach) {
   std::shared_ptr<LogicalOperator> foreach = std::make_shared<plan::Foreach>(nullptr, nullptr, nullptr, x);
   CheckPlanType(foreach.get(), RWType::RW);
 }
+
+TEST_F(ReadWriteTypeCheckTest, CheckUpdateType) {
+  RWType scenarios[16][3] = {
```

Comment:
Please use `std::array` which is [the modern version](https://en.cppreference.com/w/cpp/container/array) of "C-style" arrays. The following should work if I am not mistaken, but I might be wrong.

```suggestion
  std::array<std::array<RwType,3>, 16> scenarios = {
```